prusti-viper/src/encoder/procedure_encoder.rs

```diff
@@ -2569,66 +2594,55 @@ impl<'p, 'v: 'p, 'tcx: 'v> ProcedureEncoder<'p, 'v, 'tcx> {
         self.slice_created_at.insert(location, encoded_lhs);
 
         // TODO: what do we actually do here? this seems a littly hacky.
-        let (start, end) = match &*idx_ident {
-            "std::ops::Range" => {
-                // there's fields like _5.f$start.val_int on `encoded_idx`, it just feels hacky to
-                // manually re-do them here when we probably just encoded the type and the
-                // construction of the fields..
-                let usize_ty = self.encoder.env().tcx().mk_ty(ty::TyKind::Uint(ty::UintTy::Usize));
-                let start = encoded_idx.clone()
-                    .field(self.encoder.encode_struct_field("start", usize_ty).with_span(span)?);
-                let start_expr = self.encoder.encode_value_expr(start, usize_ty).with_span(span)?;
-
-                let end = encoded_idx
-                    .field(self.encoder.encode_struct_field("end", usize_ty).with_span(span)?);
-                let end_expr = self.encoder.encode_value_expr(end, usize_ty).with_span(span)?;
-
-                (start_expr, end_expr)
-            },
-            // other things like RangeFull, RangeFrom, RangeTo, RangeInclusive could be added here
-            // relatively easily
-            _ => todo!("{}", idx_ident),
+        // there's fields like _5.f$start.val_int on `encoded_idx`, it just feels hacky to
+        // manually re-do them here when we probably just encoded the type and the
+        // construction of the fields..
+        let usize_ty = self.encoder.env().tcx().mk_ty(ty::TyKind::Uint(ty::UintTy::Usize));
+        let start = match &*idx_ident {
+            "std::ops::Range" |
+            "std::ops::RangeFrom" => self.encoder.encode_struct_field_value(encoded_idx.clone(), "start", usize_ty)?,
+            // RangeInclusive is wierdly differnet to all of the other Range*s in that the struct fields are private
+            // and it is created with a new() fn and start/end are accessed with getter fns
+            // See https://github.com/rust-lang/rust/issues/67371 for why this is the case...
+            "std::ops::RangeInclusive" => return Err(
+                EncodingError::unsupported("RangeInclusive currently not supported".to_string())
```

Comment:
`RangeInclusive` might not be familiar to the user if they just used the shorthand syntax.

```suggestion
                EncodingError::unsupported("slicing with RangeInclusive (e.g. [x..=y]) currently not supported".to_string())
```